src/prefect/engine/result.py

```diff
@@ -123,6 +123,23 @@ def store_safe_value(self) -> None:
                 value=value, result_handler=self.result_handler
             )
 
+    def validate(self) -> bool:
+        """
+        Run any validator functions associated with this result and return whether the result is valid or not.
+
+        Returns:
+            - bool: whether or not the Result passed all validation functions
+        """
+        if self.run_validators and self.validators:
+            for validation_fn in self.validators:
+                is_valid = validation_fn(self)
+                if not is_valid:
+                    # once a validator is found to be false, this result is invalid
```

Comment:
Again with the assumptions, but what I've got here thinks ANY invalid validators means that the result is not valid.

Reply:
I'm on board with this, definitely seems like an all or nothing kind of validation step to me (i.e. you opt-in to providing X number of validators therefore you are saying they all have to be valid)

Reply:
Agreed, any invalid validator result makes the most sense. Besides, if the user wants another kind of boolean operator here (instead of AND) then they can do the same thing by combining multiple validators into a single validator that does the operator they want (like OR).